File: backend/internal/domain/mistake.go
package domain

import "time"

type MistakeType string

const (
	MistakeTypeQuantityMismatch MistakeType = "quantity_mismatch"
	MistakeTypePriceMismatch    MistakeType = "price_mismatch"
	MistakeTypeDateMismatch     MistakeType = "date_mismatch"
	MistakeTypeStatusMismatch   MistakeType = "status_mismatch"
	MistakeTypeMissingEvidence  MistakeType = "missing_evidence"
)

type Severity string

const (
	SeverityCritical Severity = "critical"
	SeverityHigh     Severity = "high"
	SeverityMedium   Severity = "medium"
	SeverityLow      Severity = "low"
	SeverityHealthy  Severity = "healthy"
)

type MistakeStatus string

const (
	MistakeStatusDetected    MistakeStatus = "detected"
	MistakeStatusUnderReview MistakeStatus = "under_review"
	MistakeStatusVerified    MistakeStatus = "verified"
	MistakeStatusResolved    MistakeStatus = "resolved"
	MistakeStatusDismissed   MistakeStatus = "dismissed"
)

type Mistake struct {
	ID                   string          `json:"id"`
	TenantID             string          `json:"tenant_id"`
	MistakeType          MistakeType     `json:"mistake_type"`
	Severity             Severity        `json:"severity"`
	Status               MistakeStatus   `json:"status"`
	AffectedEntityType   string          `json:"affected_entity_type,omitempty"`
	AffectedEntityID     string          `json:"affected_entity_id,omitempty"`
	AffectedEntityName   string          `json:"affected_entity_name,omitempty"`
	ReferenceNumber      string          `json:"reference_number,omitempty"`
	FinancialImpactMinor int64           `json:"financial_impact_minor"`
	Currency             string          `json:"currency"`
	Confidence           float64         `json:"confidence"`
	Explanation          string          `json:"explanation"`
	RecommendedAction    string          `json:"recommended_action,omitempty"`
	AssignedTo           string          `json:"assigned_to,omitempty"`
	AssignedToName       string          `json:"assigned_to_name,omitempty"`
	EvidenceIDs          []string        `json:"evidence_ids,omitempty"`
	DetectedAt           time.Time       `json:"detected_at"`
	ResolvedAt           *time.Time      `json:"resolved_at,omitempty"`
	CreatedAt            time.Time       `json:"created_at"`
	UpdatedAt            time.Time       `json:"updated_at"`
}

type MistakeTransition struct {
	ID         string        `json:"id"`
	MistakeID  string        `json:"mistake_id"`
	TenantID   string        `json:"tenant_id"`
	FromStatus MistakeStatus `json:"from_status"`
	ToStatus   MistakeStatus `json:"to_status"`
	ChangedBy  string        `json:"changed_by"`
	Reason     string        `json:"reason,omitempty"`
	CreatedAt  time.Time     `json:"created_at"`
}

File: web/src/types/mistake.ts
export type MistakeType =
  | "quantity_mismatch"
  | "price_mismatch"
  | "date_mismatch"
  | "status_mismatch"
  | "missing_evidence"
  | "lead_time_anomaly";

export type MistakeSeverity = "critical" | "high" | "medium" | "low";

export type MistakeStatus =
  | "detected"
  | "under_review"
  | "verified"
  | "resolved"
  | "dismissed";

export interface BoundingBox {
  page_number: number;
  top: number;
  left: number;
  width: number;
  height: number;
}

export interface EvidenceRef {
  id: string;
  document_id: string;
  document_name: string;
  document_type: string;
  field_name: string;
  extracted_value: string;
  confidence: number;
  bounding_box?: BoundingBox;
  raw_snippet?: string;
  observed_at: string;
}

export interface MathProof {
  formula: string;
  unit_price_minor: number;
  expected_quantity: number;
  actual_quantity: number;
  quantity_delta: number;
  expected_amount_minor: number;
  actual_amount_minor: number;
  financial_impact_minor: number;
}

export interface MistakeTransition {
  id: string;
  mistake_id: string;
  from_status: MistakeStatus;
  to_status: MistakeStatus;
  user_id: string;
  user_name: string;
  reason?: string;
  created_at: string;
}

export interface Mistake {
  id: string;
  tenant_id: string;
  title: string;
  type: MistakeType;
  severity: MistakeSeverity;
  status: MistakeStatus;
  financial_impact_minor: number;
  entity_id: string;
  entity_name: string;
  assigned_to_user_id?: string;
  assigned_to_name?: string;
  explanation: string;
  remediation_advice: string[];
  confidence_score: number;
  compound_group_id?: string;
  is_compound?: boolean;
  math_proof?: MathProof;
  evidence_items: EvidenceRef[];
  transitions?: MistakeTransition[];
  detected_at: string;
  updated_at?: string;
}
